File: carsWithAllure/src/main/java/framework/utils/Reader.java
package framework.utils;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

public class Reader {

    public static String getParametr(String parametr) {
        Properties property = new Properties();
        try {
            FileInputStream fileInputStream = new FileInputStream("src/main/resources/config.properties");
            property.load(fileInputStream);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return property.getProperty(parametr);
    }

    public static String getNameDownloadFile() {
        MyLogger.info("Получение имени файла в зависимости от ОС");
        if (System.getProperty("os.name").toLowerCase().contains("windows")) {
            return getParametr("pathToFileWindows");
        } else if (System.getProperty("os.name").toLowerCase().contains("linux")) {
            return getParametr("pathToFileLinux");
        } else {
            MyLogger.warn("Не поддерживаемая ОС");
            throw new IllegalArgumentException("Не поддерживаемая ОС");
        }
    }
}

File: ButtleShip/src/main/kotlin/framework/utils/Reader.kt
package framework.utils

import java.io.FileInputStream
import java.util.*

private const val pathToConfig: String = "src/main/resources/config.properties"
val logger: Logger = Logger()

fun getParameter(parameter: String?): String? {
    logger.info("Get $parameter  from config file")
    val property = Properties()
    property.load(FileInputStream(pathToConfig))
    return property.getProperty(parameter)
}
